src/manifestoo/addon_sorter.py

```diff
@@ -0,0 +1,44 @@
+from typing import Dict, Iterable, Set
+
+import typer
+from graphlib import TopologicalSorter
+
+from manifestoo_core.addons_set import AddonsSet
+
+from . import echo
+
+
+class AddonSorter:
+    @staticmethod
+    def from_name(name: str) -> "AddonSorter":
+        if name == "alphabetical":
+            return AddonSorterAlphabetical()
+        elif name == "topological":
+            return AddonSorterTopological()
+        else:
+            echo.error(f"Unknown sorter {name}", err=False)
+            raise typer.Exit(1)
+
+    def sort(
+        self, addons_selection: Iterable[str], addon_set: AddonsSet
+    ) -> Iterable[str]:
+        return addons_selection
+
+
+class AddonSorterAlphabetical(AddonSorter):
+    def sort(
+        self, addons_selection: Iterable[str], addon_set: AddonsSet
+    ) -> Iterable[str]:
+        return sorted(addons_selection)
+
+
+class AddonSorterTopological(AddonSorter):
+    def sort(
+        self, addons_selection: Iterable[str], addon_set: AddonsSet
+    ) -> Iterable[str]:
+        result_dict: Dict[str, Set[str]] = {}
+        for addon_name in addons_selection:
+            addon = addon_set.get(addon_name)
```

Comment:
What should happen if addon_name is not in the addon_set? Probably it cannot happen, so better fail hard in that case.

```suggestion
            addon = addon_set[addon_name]
```